### PR TITLE
Leaflet Map of Checklists

### PR DESCRIPTION
**Pull request by MuchQuak**
Pull Request Checklist:

- [x] Features and backlog bugs should be merged into the `Development` branch, **NOT** `master`
- [x] Hotfixes should be merged into both the `Development` and `master` branches (at the same time).
- [x] All new text is preferrably internationalized (i.e., no end-user-visible text is hard-coded on the PHP pages)
- [x] There are no linter errors
- [x] New features have responsive design (i.e., look aesthetically pleasing both full screen and with small or mobile screens)
- [x] [Symbiota coding standards](https://docs.google.com/document/d/1-FwCZP5Zu4f-bPwsKeVVsZErytALOJyA2szjbfSUjmc/edit?usp=sharing) have been followed
- [x] If any files have been reformatted (e.g., by an autoformatter), the reformat is its own, separate commit in the PR
- [x] Comment which GitHub issue(s), if any does this PR address
- [x] It is the code author's responsibility to merge their own pull request after it has been approved
- [x] If this PR represents a merge into the `Development` branch, remember to use the **squash & merge** option
- [x] If this PR represents a merge from the `Development` branch into the master branch, remember to use the **merge** option
- [x] If the dev team has agreed that this PR represents the last PR going into the Development branch before a tagged release (i.e., before an imminent merge into the master branch), make sure to notify the team and [lock the `Development` branch](https://github.com/BioKIC/Symbiota/settings/branches) to prevent accidental merges while QA takes place

Thanks for contributing and keeping it clean!
## Get Running
`$LEAFLET = FALSE` or `$LEAFLET = TRUE` in Symbini

## Files Changed 
`checklists/clgmap.php`

## Map Features to Check
- Can map open
  - Go to /checklists/index.php and click on the map button near a project name
- There is a marker for each checklist and titles are correct
- Double click navigates to the checklist view
- Google and leaflet functionality are the same
- Map fits to the markers so that they are in view 

## Work that was done
- Refactoring Google version of the code to easily swap with symbini flag
  - Seen in the googleInit function
- Full write up of the leaflet version of the code
  - Seen in the leafletInit function

### CODE REVIEW

File: checklists/clgmap.php [hunk 1]
```diff
@@ -15,78 +15,105 @@
 $clManager->setProj($pid);
 ?>
 <html>
-	<head>
+   <head>
+      <?php 
+        include_once($SERVER_ROOT.'/includes/leafletMap.php');
+	     include_once($SERVER_ROOT.'/includes/googleMap.php');
```
Comment (Atticus29) [inline checklists/clgmap.php:21]: Indentation is wonk here.

File: checklists/clgmap.php [hunk 2]
```diff
@@ -15,78 +15,105 @@
...
+
+         function navigateToCheckList(clid, pid) { 
+            window.location.href = `../ident/key.php?clid=${clid}&pid=${pid}&taxon=All+Species`;
+         }
+
+         function leafletInit() {
+            let map = new LeafletMap('map_canvas');
+            const markers = [];
+
+            for(let checklistId of Object.keys(checklists)) {
```
Comment (Atticus29) [inline checklists/clgmap.php:40]: TIL for... in vs for... of lol. Thanks.

File: checklists/clgmap.php [hunk 3]
```diff
@@ -15,78 +15,105 @@
...
+                  .on('dblclick', () => navigateToCheckList(checklistId, pid)));
+            }
+            const markerGroup = L.featureGroup(markers).addTo(map.mapLayer);
+            map.mapLayer.fitBounds(markerGroup.getBounds());
+         }
+
+         function googleInit() {
+
+            let map = new GoogleMap('map_canvas');
+		      let bounds = new google.maps.LatLngBounds();
```
Comment (Atticus29) [inline checklists/clgmap.php:57]: Wonky indentation?

File: checklists/clgmap.php [hunk 4]
```diff
@@ -15,78 +15,105 @@
...
-				if(minLat == maxLat && minLng == maxLng){
-					map.setCenter(swLatLng);
-				}
-				else{
-					var llBounds = new google.maps.LatLngBounds(swLatLng, neLatLng);
-					map.fitBounds(llBounds);
-				}
-			}
+            //Load Server Data from HTML Data Attributes
+            $('.service-container').each(function() {
```
Comment (Atticus29) [inline checklists/clgmap.php]: I noticed this in your other PR as well. I think that there's a general desire to move away from jQuery, but if there is no easy workaround, I personally have no issue this with.

**Review by Atticus29 — APPROVED** [review]
LGTM. Some spacing/indentation issues, and a comment about jQuery. But it works as expected and makes sense to me. Well done!

File: checklists/clgmap.php (hunk 4 above)
Comment (MuchQuak) [reply checklists/clgmap.php]: Okay, I will see if I can use something else here like document selection

File: checklists/clgmap.php (hunk 3 above)
Comment (MuchQuak) [reply checklists/clgmap.php:57]: Haha, I think my editor freaks out sometimes when doing js in php file

File: checklists/clgmap.php (hunk 4 above)
Comment (MuchQuak) [reply checklists/clgmap.php]: Okay no more jquery :)

**Pull request merged**